.github/workflows/e2e.yaml

```diff
@@ -10,6 +10,10 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
 
+env:
+  CURRENT_KIND_VERSION: 0.20.0
+  LEGACY_KIND_VERSION: 0.17.0  # for K8s versions latest kind doesn't support anymore  
+
```

Comment:
Instead of adding these to the environment like this, we could use the matrix.include functionality of github actions. https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations

This would let us specify exactly KIND version each kube version should be using.

Reply:
I considered doing that, but given that:
1. There would only ever be two kind versions
2. There's less duplication this way, you only change the kind version in one place if you want to upgrade

I decided to do it this way. I don't have a strong opinion on it though, the matrix approach is fine.